backend/app/app/tools/check_subscription.py

```diff
@@ -1,8 +1,13 @@
 import logging
 import time
+from typing import Optional
 from app.crud import crud_mongo
 
+
 def check_expiration_time(expire_time):
```

Comment:
```suggestion
def check_expiration_time(expire_time: Union[datetime, str]) -> bool:
```

Then inside the function if the type is `str` convert to a datetime by parsing, and make datetime comparisons.